internal/workload/podman/podman.go

```diff
@@ -312,16 +337,51 @@ func (p *podman) UpdateSecret(name, data string) error {
 	return p.CreateSecret(name, data)
 }
 
-func (p *podman) GenerateSystemdService(podName string, monitoringInterval uint) (service.Service, error) {
-	useName := true
-	report, err := generate.Systemd(p.podmanConnection, podName+service.PodSuffix, &generate.SystemdOptions{RestartSec: &monitoringInterval, UseName: &useName})
-	if err != nil {
-		return nil, err
+func hasAutoUpdateEnabled(labels map[string]string) bool {
+	for label := range labels {
+		if strings.Contains(label, "io.containers.autoupdate") {
+			return true
+		}
 	}
 
-	svc, err := service.NewSystemd(podName, service.ServicePrefix, report.Units)
-	if err != nil {
-		return nil, err
+	return false
+}
+
+func (p *podman) GenerateSystemdService(workload *v1.Pod, manifestPath string, monitoringInterval uint) (service.Service, error) {
+	var svc service.Service
+	podName := workload.Name
+
+	// Since podman don't support generation of the systemd services that re-creates the pod, instead of restarting it,
```

Comment:
Have you created an issue to podman for it?

Reply:
They seem to not support it: https://github.com/containers/podman/issues/13211
I think they don't want to create to complex code for generation, users can create their own services. I think this comment explain it well: https://github.com/containers/podman/blob/main/pkg/systemd/generate/pods.go#L293